### PR TITLE
Ashley Plasket

### PR DESCRIPTION
**Pull request by aplasket**
If I had more time, I would refactor parks data to be in a hash and update all methods accordingly. And work on iteration 4.

### CODE REVIEW

File: lib/park.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  def passengers
+    @vehicles.map do |vehicle|
+      vehicle.passengers
+    end.flatten
+  end
+
+  def revenue
+    park_revenue = 0
+    @vehicles.map do |vehicle|
+      park_revenue = vehicle.num_adults * @admission_price
```
Comment (dcoleman21) [inline lib/park.rb:25]: Currently this method does not pass because on line 25 you are reassigning `park_revenue` by using the `=` operator. So every time your loop runs it just over writes the last value. You will need to update that operator to `+=` so that it will add and reassign. I gave you the point for this anyway.

File: lib/passenger.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+              :age, 
+              :driver
+  def initialize(data)
+    @name = data["name"]
+    @age = data["age"]
+    @driver = false
+  end
+
+  def adult?
+    return true if age >= 18
```
Comment (dcoleman21) [inline lib/passenger.rb:12]: This method will return a `boolean` value because you are using a conditional operator `>=` so you don't need an `if` or the keywords `true` or `false`. You can just have
```
@age >= 18
```

File: lib/passenger.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    false
+  end
+
+  def driver?
+    @driver
+  end
+
+  def drive
+    @driver = true
```
Comment (dcoleman21) [inline lib/passenger.rb:21]: If you really want to make this method solid, you can add this
```
@driver = true if adult?
```
Which will only allow a passenger to become a driver, when the `.drive` method is called, if they are an adult.

File: lib/vehicle.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
+class Vehicle
+  attr_reader   :year,
+                :name,
+                :make, 
+                :speeding
+  attr_accessor :passengers
+  
+  def initialize(year, name, make)
```
Comment (dcoleman21) [inline lib/vehicle.rb:8]: Really make sure you are paying attention to the interaction pattern:
```
pry(main)> vehicle = Vehicle.new("2001", "Honda", "Civic")    
# => #<Vehicle:0x00007fe0da9c63d8...>

pry(main)> vehicle.year
# => "2001"

pry(main)> vehicle.make
# => "Honda"

pry(main)> vehicle.model
# => "Civic"
```
The attributes are `year`, `make`, `model`. So as this is now it does not pass because your variable `name` is storing the value of `make` and your variable `make` is storing the value of `model`. Put a pry on line 8 1/2 and see what each of those is holding. Remember that you will also need to change the order of things on lines 9 - 11 and in your `attr_reader`. I did give you this point.

File: lib/vehicle.rb [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    @speeding = true
+  end
+
+  def add_passenger(passenger)
+    @passengers << passenger
+  end
+
+  def num_adults
+    count_adults = 0
+    @passengers.map do |passenger|
```
Comment (dcoleman21) [inline lib/vehicle.rb:30]: Look into the `count` enumerable.

File: spec/vehicle_spec.rb [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+  end
+
+  describe "#initialize" do
+    it 'exists' do
+      expect(@vehicle).to be_a(Vehicle)
+    end
+
+    it 'has attributes' do
+      expect(@vehicle.year).to eq("2001")
+      expect(@vehicle.name).to eq("Honda")
```
Comment (dcoleman21) [inline spec/vehicle_spec.rb:15]: Make sure to follow the interaction pattern closely:
```
pry(main)> vehicle = Vehicle.new("2001", "Honda", "Civic")    
# => #<Vehicle:0x00007fe0da9c63d8...>

pry(main)> vehicle.year
# => "2001"

pry(main)> vehicle.make
# => "Honda"

pry(main)> vehicle.model
# => "Civic"
```
`name` is not one of the attributes.

File: spec/vehicle_spec.rb [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+      expect(@vehicle.make).to eq("Civic")
+      expect(@vehicle.speeding).to eq(false)
+      expect(@vehicle.passengers).to eq([])
```
Comment (dcoleman21) [inline spec/vehicle_spec.rb:18]: I would put this in its own test
```
it 'starts with no passengers' do 
```

**Review by dcoleman21 — COMMENTED** [review]
Overall, nice job. You are making great progress. I left some comments for you to review. Let me know if you have any questions.

File: lib/park.rb (hunk 1 above)
Comment (aplasket) [reply lib/park.rb:25]: @dcoleman21 that makes sense, had I tried to add a second vehicle to one of the parks def would have had a failed spec test!

File: lib/passenger.rb (hunk 3 above)
Comment (aplasket) [reply lib/passenger.rb:21]: hey @dcoleman21 totally makes sense. curious though, I don't see in interaction pattern or description that a person needs to be an adult in order to drive. Would you mind pointing out to me where I can find that  in the pattern? Or was that an inference I was supposed to figure out based on the adult? method being called from Charlie and then Taylor

File: lib/vehicle.rb (hunk 4 above)
Comment (aplasket) [reply lib/vehicle.rb:8]: @dcoleman21 ahhh shoot, thanks for calling this out!

File: lib/vehicle.rb (hunk 5 above)
Comment (aplasket) [reply lib/vehicle.rb:30]: cool thanks for the tip!

File: spec/park_spec.rb [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+    it 'exists' do
+      expect(@forest_park).to be_a(Park)
+      expect(@yo_yo_park).to be_a(Park)
+    end
+
+    it 'has attributes' do
+      expect(@forest_park.name).to eq('forest park')
+      expect(@forest_park.admission_price).to eq(20)
+      expect(@yo_yo_park.admission_price).to eq(5)
+      expect(@forest_park.vehicles).to eq([])
```
[the remark this answers is not in the document]
Comment (aplasket) [reply spec/park_spec.rb:19]: @dcoleman21  that makes sense! is it ok that vehicles is listed as an instance variable, or in this case should it be defined as a method?

File: lib/passenger.rb (hunk 3 above)
Comment (dcoleman21) [reply lib/passenger.rb:21]: @aplasket You are correct, I don't think that it explicitly stated anything about needing to be an adult to drive. Just added this here as a suggestion for thinking about edge case testing. If you were to add to your #drive test a case where a passenger that was not an adult tried to drive, that guard statement should catch it. But just an extra insight at this point.
